### PR TITLE
Common: Fix Recursive CreateDirectoryPath() on Windows

### LINKED ISSUE #11784: [BUG]: FileSystem: recursive CreateDirectoryPath with absolute path broken on Windows
### Describe the Bug

If `FileSystem::CreateDirectoryPath` is called in the following situation

* on Windows
* with an absolute path (eg. `C:\...`)
* with recursive=true
* and the parent directory does not exist yet (ie. recursive is necessary)

then it will always fail to create the directory because of a bug in the [recursive implementation](https://github.com/PCSX2/pcsx2/blob/7d6f74083ba656f7de0a6a2148be4851bf987620/common/FileSystem.cpp#L1726-L1777).

The way recursive CreateDirectoryPath works is given `C:\aaa\bbb`, it will attempt to create the following sequence with CreateDirectoryW

```
C:
C:\aaa
C:\aaa\bbb
```

However for `C:`, CreateDirectoryW fails with `ERROR_PATH_NOT_FOUND` instead of what the code is expecting, `ERROR_ALREADY_EXISTS`. (Note that `C:\`, `\\?\C:`, and `\\?\C:\` all produce the same result.) This causes it to stop at the first step with an error.

I have a [work-around](https://github.com/PCSX2/pcsx2/commit/854e82e53d6ca0ecf70832734456fc0196a10902) that skips a prefix ending with `:\`. This is working for me, but I don't know how robust it is.

### Reproduction Steps

I found this while working on a branch where I called CreateDirectoryPath. I don't know if there is a way to actually trigger this bug from within main PCSX2, so if you don't want to track this without a repro, you can close it.

AFAICT CreateDirectoryPath is called with recursive=true only twice in the codebase, once in [QtHost.cpp](https://github.com/PCSX2/pcsx2/blob/7d6f74083ba656f7de0a6a2148be4851bf987620/pcsx2-qt/QtHost.cpp#L1596) and once in [MemoryCardFolder.cpp](https://github.com/PCSX2/pcsx2/blob/7d6f74083ba656f7de0a6a2148be4851bf987620/pcsx2/SIO/Memcard/MemoryCardFolder.cpp#L2000). I don't know whether these callsites are vulnerable or not.

### Expected Behavior

_No response_

### PCSX2 Revision

f1abee5d0ba5af70454b1c956bce42060d3cd98e

### Operating System

Windows 10 (64bit)

### If Linux - Specify Distro

_No response_

### Logs & Dumps

_No response_

### PR DESCRIPTION
**Pull request by TheLastRar**
### Description of Changes
Skip over the root of the path when creating directories recursively on Windows.

### Rationale behind Changes
The existing logic would call `CreateDirectoryW()` when it encounters a directory separator
As the path would something like `\\?\C:\foo\bar` we would try to create a directory using an empty string.

As mentioned here https://github.com/PCSX2/pcsx2/issues/11784, only two locations use this,
Once in [QtHost.cpp](https://github.com/PCSX2/pcsx2/blob/7d6f74083ba656f7de0a6a2148be4851bf987620/pcsx2-qt/QtHost.cpp#L1596), which itself is only used to download fonts and currently doesn't need recursive directory creation.
Once in [MemoryCardFolder.cpp](https://github.com/PCSX2/pcsx2/blob/7d6f74083ba656f7de0a6a2148be4851bf987620/pcsx2/SIO/Memcard/MemoryCardFolder.cpp#L2000), which I think gets called when PS2 games open/create files on the (folder) memory card. Unsure if any games actually need recursive behaviour here.

Fixes #11784 

### Suggested Testing Steps
@scurest can you give this a test

### CODE REVIEW

File: common/FileSystem.cpp [hunk 1]
```diff
@@ -1732,8 +1732,26 @@ bool FileSystem::CreateDirectoryPath(const char* Path, bool Recursive, Error* er
...
+		// for absolute paths, we need to skip over the path root
+		size_t rootLength = 0;
+		if (Path::IsAbsolute(Path))
+		{
+			const wchar_t* root_start = wpath.c_str();
+			wchar_t* root_end;
+			const HRESULT hr = PathCchSkipRoot(const_cast<wchar_t*>(root_start), &root_end);
+			if (FAILED(hr))
+			{
+				Console.ErrorFmt("PathCchSkipRoot() returned {:08X}", static_cast<unsigned>(hr));
```
Comment (scurest) [inline common/FileSystem.cpp:1744]: Should this use Error::SetHResult instead of logging to console?

**Comment by scurest** [comment]
Thanks! I don't have a Windows dev machine to test the whole thing, but this fragment appears to be skipping the correct prefix in some Github action tests.

File: common/FileSystem.cpp (hunk 1 above)
Comment (TheLastRar) [reply common/FileSystem.cpp]: Yes it should, good catch

**Comment by F0bes** [comment]
Are you happy with this @TheLastRar ?

**Comment by TheLastRar** [comment]
Yeah, did some local testing on it myself as well and it worked.

**Pull request merged**